src/common.jl

```diff
@@ -123,6 +124,145 @@ function solve{uType,tType,isinplace,algType<:SundialsAlgorithm,F}(
         end
     end
 
-    build_ode_solution(prob,alg,ts,timeseries,
+    build_solution(prob,alg,ts,timeseries,
+                      timeseries_errors = timeseries_errors)
+end
+
+## Solve for DAEs uses IDA
+
+function solve{uType,duType,tType,isinplace,algType<:SundialsDAEAlgorithm,F}(
+    prob::AbstractDAEProblem{uType,duType,tType,isinplace,F},
+    alg::Type{algType},timeseries=[],ts=[],ks=[];
+    callback=()->nothing,abstol=1/10^6,reltol=1/10^3,
+    saveat=Float64[],adaptive=true,maxiter=Int(1e5),
+    timeseries_errors=true,save_timeseries=true,
+    userdata=nothing,kwargs...)
+
+    tspan = prob.tspan
+    t0 = tspan[1]
+    T = tspan[end]
+
+    save_ts = sort(unique([t0;saveat;T]))
+
+    if T < save_ts[end]
+        error("Final saving timepoint is past the solving timespan")
+    end
+    if t0 > save_ts[1]
+        error("First saving timepoint is before the solving timespan")
+    end
+
+    if typeof(prob.u0) <: Number
+        u0 = [prob.u0]
+    else
+        u0 = vec(deepcopy(prob.u0))
+    end
+
+    if typeof(prob.du0) <: Number
+        du0 = [prob.du0]
+    else
+        du0 = vec(deepcopy(prob.du0))
+    end
+
+    sizeu = size(prob.u0)
+    sizedu = size(prob.du0)
+
+    ### Fix the more general function to Sundials allowed style
+    if !isinplace && (typeof(prob.u0)<:Vector{Float64} || typeof(prob.u0)<:Number)
+        f! = (t,u,du,out) -> (out[:] = prob.f(t,u,du); 0)
+    elseif !isinplace && typeof(prob.u0)<:AbstractArray
+        f! = (t,u,du,out) -> (out[:] = vec(prob.f(t,reshape(u,sizeu),reshape(du,sizedu))); 0)
+    elseif typeof(prob.u0)<:Vector{Float64}
+        f! = prob.f
+    else # Then it's an in-place function on an abstract array
+        f! = (t,u,du,out) -> (prob.f(t,reshape(u,sizeu),reshape(du,sizedu),out);
+                          u = vec(u); du=vec(du); 0)
+    end
+
+    mem = IDACreate()
+    if mem == C_NULL
+        error("Failed to allocate IDA solver object")
+    end
+
+    ures = Vector{Vector{Float64}}()
+    ts   = [t0]
+
+    try
+        userfun = UserFunctionAndData(f!, userdata)
+        u0nv = NVector(u0)
+        flag = @checkflag IDAInit(mem, cfunction(idasolfun,
+                                  Cint, (realtype, N_Vector, N_Vector,
+                                  N_Vector, Ref{typeof(userfun)})),
+                                  t0, convert(N_Vector, u0),
+                                  convert(N_Vector, du0))
+        flag = @checkflag IDASetUserData(mem, userfun)
+        flag = @checkflag IDASStolerances(mem, reltol, abstol)
+        flag = @checkflag IDASetMaxNumSteps(mem, maxiter)
+        flag = @checkflag IDADense(mem, length(u0))
+
+        push!(ures, copy(u0))
+        utmp = NVector(copy(u0))
+        dutmp = NVector(copy(u0))
+        tout = [0.0]
+
+        rtest = zeros(length(u0))
+        f!(t0, u0, du0, rtest)
+        if any(abs(rtest) .>= reltol)
```

Comment:
Good catch. While this would've worked in v0.5 and v0.6, might as well fix it right now. I fixed it.